File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.py
def solution(num_list):
    answer = 0
    
    tmp1 = 1
    tmp2 = 0
    
    for num in num_list:
        tmp1 *= num
    
    for num in num_list:
        tmp2 += num
        
    tmp2 *= tmp2
    
    if tmp1 < tmp2:
        answer = 1
    else :
        answer = 0
        
    return answer

File: 프로그래머스/0/181929. 원소들의 곱과 합/원소들의 곱과 합.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> num_list) {
    int answer = 0;
    int num_list_len = num_list.size();
    int tmp1 = 1;
    int tmp2 = 0;
    
    // 1. 모든 원소의 곱 계산
    for (size_t i = 0; i < num_list_len; i++) {
        tmp1 *= num_list[i];
    }
    
    // 2. 모든 원소의 합 계산
    for (size_t i = 0; i < num_list_len; i++) {
        tmp2 += num_list[i];
    }
    
    // 3. 합의 제곱 계산
    tmp2 *= tmp2;
    
    // 4. 조건 비교 후 결과 저장
    if (tmp1 < tmp2) {
        answer = 1;
    } else {
        answer = 0;
    }
    
    return answer;
    return answer;
}